tools/site_and_regional/neon_surf_wrapper.py

```diff
@@ -52,6 +52,12 @@ def get_parser():
                 dest="verbose", 
                 default=False)
 
+    parser.add_argument('-f', '--fates',
+                        help='Create and/or modify surface data file for FATES. ',
```

Comment:
Sounds good to me.

Reply:
Actually here we may want to leave the FATES signifier since we (in my opinion) do things FATES-specific like not choosing one dominant PFT. I can change to a `--16pft` flag for the `modify_singlept_site_neon` script though?

Reply:
Ahhh, good point. Hmmm. Although I still think the standard thing would be to do the dominate PFT. Full FATES will ignore that so that's sufficient for that and cases you want to replicate the BGC setup. It's the "in between" modes that someone might want to have multiple PFT's where it matters here.

Reply:
Should we have a quick call to discuss?

Reply:
> Should we have a quick call to discuss?

Yes lets try to include @jkshuman as well?

Reply:
I know Jackie is busy this week. She's trying to get a manuscript done. We can always ask though.

Reply:
@ekluzek just met briefly. The issue is that the 1 degree surface files are "wrong" with respect to what the NEON data says is actually present at the site, but right now we only have data on 1 dominant PFT per site. So for now we **are** going to just go with 1 dominant PFT for the surface files. Remember that in full FATES mode you can still have multiple PFTs, based on your FATES parameter file. 

In the future we would like to update the subset surface data file to have the PFT mixture match what is observed at the NEON site (using inventory data, etc.). I will make an issue about this.

Reply:
@ekluzek where is the dominant PFT information coming from if the surface data files are wrong? Does it make sense to wait until https://github.com/ESCOMP/CTSM/issues/1934 is implemented so that we can run mixed PFT FATES simulations?

Reply:
Here's what @adrifoster and I thought at our discussion yesterday. First the dominant PFT information IS correct. It is coming from NEON, and we think it's all correct. What we currently don't have are the inventory files, nor a list of the PFT's that are at the site. So implementing #1934 will take both software infrastructure work, as well as science research to express inventory data into % cover.

The part that would be wrong is to use the 1-degree mix of PFT's from the global 1-degree dataset and say that represents the site. We could turn off dominant PFT and use the 1-degree mix, and for some sites that might be OK. But, I'd expect it to be wrong and too extensive for most sites. We could do a check and see what the list is for the sites when run this way though -- maybe I'm wrong and the list is reasonable if not ideal. It's just when I look out at the world a local small site normally has a really restricted set of PFT's, while a large 1-degree grid-cell is going to have tons of variety.

Since, it will take time to do #1934 we figured starting with the dominant PFT makes sense. For full FATES you'll still get whatever mix of PFT's you want to have, so it's fine for full FATES. It's also fine for FATES-SP mode where you just want to compare to CTSM-BGC fro the simplest possible case. What it is missing is the complexity cascade of going from FATES-SP, to fixed biogeography, to NOCOMP, to full FATES where you really want to have the mix of PFT's. But, in terms of getting something in place now it's a nice baby step. It also means that people can readily run FATES for NEON sites and at least get their feet wet. You are right to take a full dive, we need #1934, but as it's going to be complex enough and take enough time -- we figure doing a baby step would be the right thing to do.

We could do a group discussion if that helps? Maybe next week?

Reply:
@ekluzek I ask about this b/c at the SE meeting yesterday we had all agreed that the implementation was to run FATES with mixed PFTs as this allows for testing across modes. The mix of PFTs is an important piece for calibration and validation. The desire to skip the surface dataset and use this NEON specific dataset with only the dominant PFT was not introduced yesterday as a critical limiting factor for providing FATES-NEON function. This is an important point. There will be differences in calibrating to 1 PFT vs a mix of PFTs, but this PR addresses only function and not calibration. 

There are a number of ways to determine dominant PFT in a forest stand, and many of these NEON sites do in reality have multiple species and potentially multiple PFTs present in the inventory data. NEON probably has a method to provide this dominant PFT, so it would be good to have that information in this NEON file as a comment. It seems the suggestion is to now use the NEON specific dataset where the only data provided is the dominant 1 PFT. 

Then with https://github.com/ESCOMP/CTSM/issues/1934 the PFT mix will be adjusted based on reality from the inventory.

Reply:
Currently when I generate surface data for NEON sites I use this [wrapper](https://github.com/ESCOMP/CTSM/blob/master/tools/site_and_regional/neon_surf_wrapper.py) which get's dominant PFT (based on expert opinion) from this ([.csv file](https://github.com/ESCOMP/CTSM/blob/master/tools/site_and_regional/neon_sites_dompft.csv)). 

We could define additional (common) pfts in the this file to be read in for FATES surface data, but again this seems like it would take a bit longer to do well enough to be defensible. FWIW, I like the baby-step approach we're taking, with a more robust estimate of veg cover coming in with #1934.  Based on what Adrianna showed yesterday it seems like this isn't far from being ready to go NEON-wide?

Reply:
Sorry @jkshuman I really do not mean to go against our agreements from the meeting. But, a challenge we have is that even though we were both in the same meeting we all have different expertise and so pick up on different things. And we might all might miss something stated because of loss of focus. Since, I don't live in FATES I didn't pick up on the need to ensure there are multiple PFT's. What I am trying to do is to make sure we aren't doing something that struck me as wrong. I did honestly think in the meeting that we were talking about having the datasets use the dominant PFT the same as the CTSM-BGC datasets.

What I hear you saying is that you'd like to use the 1-degree gridcell PFT's for this first go around. And then later we will get the corrected PFT distribution from the NEON inventory data. Is that right? So the first baby-step is to use the 1-degree PFT's, and then with #1934 we'll get the correct ones.

It struck me as the most correct to use the dominate PFT as that's from the actual site. But, you and @adrifoster are the experts in FATES here, so it should be done in the way you think is most correct.  I think it is important to realize it is wrong either way, for this first baby step, but yes you and @adrifoster should decide which way you think it should be for this first step. The ones that are the experts in the question at hand should be the ones that decide.

So sorry for derailing the train -- I really didn't mean to do that. :-(

Reply:
Thanks @ekluzek 

I wanted to provide an example of how this looks in these forests. From an ecological perspective, it is important to include more than just 1 PFT at many of these sites. In many locations even at the individual-plot level (i.e. a few hundred meters squared) we can see a large number of species of differing PFTs. For example, looking at the inventory plots for HARV, which surround the flux tower and are intended to be representative of the forest, we can see a huge mix of species (upwards of 50 species), and even reducing to just the genus we still have quite a mix (~35 genera):

![by_genus](https://user-images.githubusercontent.com/13225250/212429040-27ea1608-387c-48ec-80dd-df68791cfaa7.png)

If we further reduce that to PFTs, we mostly get a mix of deciduous broadleaf and needleleaf evergreen trees (about 60% deciduous, and 40% evergreen). From an ecological perspective, I don't think it is appropriate to just say this site is "deciduous broadleaf", even though that is what the [csv file ](https://github.com/ESCOMP/CTSM/blob/master/tools/site_and_regional/neon_sites_dompft.csv) classifies it as.

![by_pft](https://user-images.githubusercontent.com/13225250/212429275-fb8587a1-c0ef-411c-bb2b-4d317b4c35e4.png)

Here are averages across the inventory plots for ABBY, BONA, and HARV:

![average_pft](https://user-images.githubusercontent.com/13225250/212429720-0aba60c3-0cab-4e09-8c79-733938b0589a.png)

As you can see, while ABBY is pretty dominated by NET, even the canonically "floristically simple" boreal forest at BONA is quite mixed, even though it is classified in the csv as just being needleleaf evergreen.

I think that since we are interested in understanding both forest dynamics (FATES) as well as land-surface interactions (CTSM and FATES-CTSM), we would not want to discount a whole 40% of the forest that is photosynthesizing (or not) year-round (i.e. needleleaf evergreen trees vs. deciduous broadleaf trees), and has quite a different albedo, among other differences. I think this would impact both the FATES-CTSM runs as well as the CTSM Big-Leaf runs. 

Since we do have access to the actual data on PFT composition at the NEON sites it seems quite pertinent that we use that data to drive our model runs (as planned in #1934).

Reply:
Awesome, thanks for the analysis that's helpful. The naive question I wonder about is what the PFT mix will be based on the 1 degree grid cell for each of these sites? Is it going to be about right or include PFTs that really shouldn't be there? The % cover for each is going to be wrong for sure, but is it going to include PFTs that really aren't at the local site? I was thinking that would be the likely problem, but maybe PFTs are broad enough classes that you won't have that problem....

Reply:
I think that the PFTs are mostly broad enough that it won't make a huge difference.

For example, for the 1-degree surrounding HARV:

```
 PCT_NAT_PFT =
  0,
  15.5359300255115,
  0,
  0,
  0,
  0,
  0,
  78.9754444442113,
  0,
  0,
  0,
  0,
  0,
  5.47422440732277,
  0.014401122954473 ;

```
But we should try to update it using the data ASAP.

Reply:
Thanks @ekluzek We were talking about different datasets, and now that we all have the full information we can make an informed decision.

Reply:
Resolving this